dvc/output/base.py

```diff
@@ -288,13 +288,13 @@ def save(self):
 
         self.hash_info = self.get_hash()
 
-    def commit(self):
+    def commit(self, filter_info=None):
         if not self.exists:
             raise self.DoesNotExistError(self)
-
+        path_info = filter_info if filter_info else self.path_info
         assert self.hash_info
         if self.use_cache:
-            self.cache.save(self.path_info, self.cache.tree, self.hash_info)
+            self.cache.save(path_info, self.cache.tree, self.hash_info)
```

Comment:
`self.hash_info` doesn't really correspond to the `filter_info`, we need to either use that "partial" hash for filter_info or teach `self.cache.save` use `self.hash_info` but take `filter_info` into consideration.

Reply:
Done